frontend/src/app/services/session.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SessionService {
+    private static sessionExpiredSubject = new Subject<void>();
+    public expired = SessionService.sessionExpiredSubject.asObservable();
+
+    public static markExpired() {
```

Comment:
I don't see why these properties should be marked as `static`. What sets them apart from properties in, e.g., the `AuthService`? As far as I know, static properties are independent of an instance, but since services are singletons anyway, and the `Subject` here does not retain any state, the difference is quite moot (but maybe I'm missing something!).